_hardware/cypherockx1.md

```diff
@@ -16,52 +16,138 @@ providerWebsite: https://cypherock.com/
-> - Keys can be recovered by tapping any 2 out of the 4 cyCards on the X1 Wallet. Each cyCard can also be protected by a PIN just like any other hardware wallet. The recovered keys are stored in a volatile memory on the X1 Wallet which gets wiped out as soon as the operation is done.
-> - The user can store 3 wallet recovery phrases in a single product.
-> - The cyCards can be distributed around different places to increase security of the funds and protect against $5 wrench attacks.
+### Alternatively, one can follow below commands in a *powershell* window to build the Cypherock X1 Wallet firmware:
+* For Initial firmware - unsigned binary will be generated at build/initial/Cypherock-Initial.bin
+        
+        docker build . --file Dockerfile --tag x1-wallet-app-env
+        mkdir -p build
+        docker run -v ${pwd}/build:/out -it x1-wallet-app-env /bin/ash
+        mkdir -p /dist/initial && cd /home
+        cd build/Release/
+        cmake -DDEV_SWITCH=OFF -DDEBUG_SWITCH=OFF -DSIGN_BINARY=OFF -DCMAKE_BUILD_TYPE:STRING=Release -DFIRMWARE_TYPE=Initial -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=ON -DCMAKE_BUILD_PLATFORM:STRING=Device -G Ninja ../../
+        ninja
+        cp Cypherock-*.* /dist/initial && cp -a /dist/. /out
+        exit
 
-From the [FAQ](https://www.cypherock.com/faq/): Can the company somehow steal the digital assets of the user?
+* For Main firmware - unsigned binary will be generated at build/main/Cypherock-Main.bin
+        
+        docker build . --file Dockerfile --tag x1-wallet-app-env
+        mkdir -p build
+        docker run -v ${pwd}/build:/out -it x1-wallet-app-env /bin/ash
+        mkdir -p /dist/main && cd /home
+        cd build/Release/
+        cmake -DDEV_SWITCH=OFF -DDEBUG_SWITCH=OFF -DSIGN_BINARY=OFF -DCMAKE_BUILD_TYPE:STRING=Release -DFIRMWARE_TYPE=Main -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=ON -DCMAKE_BUILD_PLATFORM:STRING=Device -G Ninja ../../
+        ninja
+        cp Cypherock-*.* /dist/main && cp -a /dist/. /out
+        exit
+
+
+### Reproducing the Cypherock X1 Wallet Initial firmware v0.4.773 on Windows 11
+* On Git Bash Shell, checkout using tag:
+        
+        MINGW64 ~/Desktop/Cypherock/X1_wallet/Firmware/v0.4.773 ((v0.4.773)) $ git clone git@github.com:Cypherock/x1_wallet_firmware.git .
+        MINGW64 ~/Desktop/Cypherock/X1_wallet/Firmware/v0.4.773 ((v0.4.773)) $ git checkout v0.4.773
+        .
+        .
+
+* On Powershell, setup Docker and compile:
+
+        PS C:\Users\LENEVO\Desktop\Cypherock\X1_wallet\Firmware\v0.4.773>  docker run -v ${pwd}/build:/out -it x1-wallet-app-env /bin/ash docker build . --file Dockerfile --tag x1-wallet-app-env
+        .
+        .
+        /home/build/Release # cmake -DDEV_SWITCH=OFF -DDEBUG_SWITCH=OFF -DSIGN_BINARY=OFF -DCMAKE_BUILD_TYPE:STRING=Release -DFIRMWARE_TYPE=Initial -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=ON -DCMAKE_BUILD_PLATFORM:STRING=Device -G Ninja ../../
+        /home/build/Release # ninja
```

Comment:
`ninja` fails for me. I had tried it first with the first  code block and now again with this one with the submodules but both times I get:

```
...
   72 | #include "lv_port_disp.h"
      |          ^~~~~~~~~~~~~~~~
compilation terminated.
ninja: job failed: /usr/bin/arm-none-eabi-gcc -DFIRMWARE_HASH_CALC=1 -DSTM32L4 -DSTM32L486xx -DUSBD_SOF_DISABLED -DUSE_BIP32_CACHE=0 -DUSE_BIP39_CACHE=0 -DUSE_HAL_DRIVER -DUSE_SIMULATOR=0 -DX1WALLET_INITIAL=1 -DX1WALLET_MAIN=0 -I../../initial/config -I../../src -I../../src/level_one/controller -I../../src/level_one/tasks -I../../src/level_two/controller -I../../src/level_two/tasks -I../../src/level_three/add_wallet/controller -I../../src/level_three/add_wallet/tasks -I../../src/level_three/advanced_settings/controller -I../../src/level_three/advanced_settings/tasks -I../../src/level_three/old_wallet/controller -I../../src/level_three/old_wallet/tasks -I../../src/level_four/core/controller -I../../src/level_four/core/tasks -I../../src/level_four/card_health_check -I../../src/level_four/factory_reset -I../../src/level_four/tap_cards/controller -I../../src/level_four/tap_cards/tasks -I../../common/interfaces/card_interface -I../../common/interfaces/desktop_app_interface -I../../common/interfaces/flash_interface -I../../common/interfaces/user_interface -I../../common/libraries/atecc -I../../common/libraries/atecc/atcacert -I../../common/libraries/atecc/basic -I../../common/libraries/atecc/crypto -I../../common/libraries/atecc/crypto/hashes -I../../common/libraries/atecc/hal -I../../common/libraries/atecc/host -I../../common/libraries/atecc/jwt -I../../common/libraries/crypto -I../../common/libraries/crypto/aes -I../../common/libraries/crypto/chacha20poly1305 -I../../common/libraries/crypto/ed25519-donna -I../../common/libraries/crypto/monero -I../../common/libraries/crypto/random_gen -I../../common/libraries/proof_of_work -I../../common/libraries/shamir -I../../common/libraries/util -I../../common/startup -I../../common/logger -I../../common/coin_support -I../../common/flash -I../../common/Firewall -I../../common -I../../common/lvgl -I../../common/lvgl/porting -I../../common/lvgl/src -I../../common/lvgl/src/lv_core -I../../common/lvgl/src/lv_draw -I../../common/lvgl/src/lv_font -I../../common/lvgl/src/lv_hal -I../../common/lvgl/src/lv_misc -I../../common/lvgl/src/lv_objx -I../../common/lvgl/src/lv_themes -I../../stm32-hal -I../../stm32-hal/BSP -I../../stm32-hal/Inc -I../../stm32-hal/Drivers/CMSIS/Include -I../../stm32-hal/Drivers/CMSIS/Device/ST/STM32L4xx/Include -I../../stm32-hal/Drivers/STM32L4xx_HAL_Driver/Inc -I../../stm32-hal/Drivers/STM32L4xx_HAL_Driver/Inc/Legacy -I../../stm32-hal/Drivers/STM32L4xx_HAL_Driver/Src -I../../stm32-hal/Middlewares/ST/STM32_USB_Device_Library/Class/CDC/Inc -I../../stm32-hal/Middlewares/ST/STM32_USB_Device_Library/Class/CDC/Src -I../../stm32-hal/Middlewares/ST/STM32_USB_Device_Library/Core/Inc -I../../stm32-hal/Middlewares/ST/STM32_USB_Device_Library/Core/Src -I../../stm32-hal/Peripherals -I../../stm32-hal/Peripherals/Buzzer -I../../stm32-hal/Peripherals/display -I../../stm32-hal/Peripherals/display/SSD1306 -I../../stm32-hal/Peripherals/flash -I../../stm32-hal/Peripherals/logger -I../../stm32-hal/Peripherals/nfc -I../../stm32-hal/Src -I../../stm32-hal/Startup -I../../stm32-hal/porting -I../../stm32-hal/libusb -I../../stm32-hal/libusb/inc -I../../stm32-hal/libusb/src -O3 -DNDEBUG -mcpu=cortex-m4 -mthumb -mfpu=fpv4-sp-d16 -mfloat-abi=hard -fdata-sections -ffunction-sections -Wall -Wno-format-truncation -Wno-unused-but-set-variable -Wno-return-type -D_POSIX_C_SOURCE=200809L -std=c11 -MD -MT CMakeFiles/Cypherock-Initial.elf.dir/src/level_one/controller/controller_level_one_initial.c.obj -MF CMakeFiles/Cypherock-Initial.elf.dir/src/level_one/controller/controller_level_one_initial.c.obj.d -o CMakeFiles/Cypherock-Initial.elf.dir/src/level_one/controller/controller_level_one_initial.c.obj -c ../../src/level_one/controller/controller_level_one_initial.c
In file included from ../../common/interfaces/user_interface/ui_common.h:13,
                 from ../../common/interfaces/user_interface/ui_menu.h:13,
                 from ../../src/constant_texts.h:13,
                 from ../../src/controller_main.h:32,
                 from ../../src/level_one/controller/controller_level_one.h:14,
                 from ../../src/level_one/controller/controller_level_one_initial.c:60:
../../common/lvgl/lvgl.h:72:10: fatal error: lv_port_disp.h: No such file or directory
   72 | #include "lv_port_disp.h"
      |          ^~~~~~~~~~~~~~~~
compilation terminated.
ninja: subcommands failed
```

I do not have a windows system but I doubt that's the issue.

Reply:
It failed due to incomplete clone of the git repository. You need to use `--recurse-submodules` switch with the git command. 
I have updated the commands.